subsys/net/lib/dns/mdns_responder.c

```diff
@@ -35,6 +35,18 @@ LOG_MODULE_REGISTER(net_mdns_responder, CONFIG_MDNS_RESPONDER_LOG_LEVEL);
 
 #include "net_private.h"
 
+/*
+ * GCC complains about struct sockaddr accesses due to the various
+ * address-family-specific variants being of differing sizes. Let's not
+ * mess with code (which looks correct), just silence the compiler.
```

Comment:
Interesting. I have not seen these issues so far, what gcc version is this?

Also could you add `net: ` prefix to the dns commit subject, it is just easier to find it later if needed.
So `net: dns: ....`

Reply:
Looks like I'm using gcc 14.2.0 on Debian unstable. I'lve fixed the dns: commit message, now to go fix more build failures.